File: static/code/sample/tethered/gdl-hidipswch-b.py
# In this sample:
# Scan all switches and indicate if any switch is in the ON position.

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def Read(index):
    ret = duelink.Engine.ExecuteCommand(f"Read({index})")
    return ret != 0

while True:
    for i in range(10):
        if Read(i + 1):
            print(f"Detected ON on switch index: {i + 1}")
    
    time.sleep(1) 

File: static/code/sample/tethered/gdl-hidipswch-b.cs
// In this sample:
// Scan all switches and indicate if any switch is in the ON position.

using System;
using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

bool Read(int index) {
    var ret = duelink.Engine.ExecuteCommand($"Read({index})");
    return ret != 0;
}

while (true) {
    for (var i = 0; i < 10; i++) {
        if (Read(i + 1)) {
            Console.WriteLine($"Detected ON on switch index: {(i + 1)}");
        }
    }
        
    Thread.Sleep(1000);
}